avocado/utils/genio.py

```diff
@@ -174,3 +174,22 @@ def write_one_line(filename, line):
     :type line: str
     """
     write_file(filename, line.rstrip('\n') + '\n')
+
+
+def safe_write_file(filename, data):
+    """
+    Write data to a file.
```

Comment:
Actually less is coming and right now. I see your usage is purely try and log the issue. I don't really like this, I think this function should just raise the exception. Anyway I understand the usecase so I'd just change the name (again, sorry for not noticing this earlier) as it's not really a `safe_write_file` as it won't make sure the data are written, it's simply tries to write and eventually logs a failure. So how about `write_file_or_log`? That would reflect the usage more clearly.